### PR TITLE
Apim 4805 api category search run e2e

### PR DESCRIPTION
**Pull request by illichso**
## Issue

https://gravitee.atlassian.net/browse/APIM-4805

## Description

Added possibility to search by category id, or category key.
Before my changes: the endpoint only allows keyword search and cannot be given a category context.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-eeosfffiep.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java [hunk 1]
```diff
@@ -122,11 +122,12 @@ public Response getApis(@BeanParam PaginationParam paginationParam, @BeanParam A
     @RequirePortalAuth
     public Response searchApis(
         @NotNull(message = "Input must not be null.") @QueryParam("q") String query,
+        @QueryParam("category") String category,
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java:125]: [This](https://gravitee.slab.com/posts/gravitee-rest-api-guidelines-8ta6qkxl) doc says not to use query params on the _search endpoints, but even though there are already similar implementations. 
Worth to check with team

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/ApisResourceTest.java [hunk 2]
```diff
@@ -326,6 +328,24 @@ public void shouldSearchApis() throws TechnicalException {
...
 
+    @Test
+    public void shouldSearchApisWithCategory() throws TechnicalException {
+        doReturn(new HashSet<>(List.of("3")))
+            .when(filteringService)
+            .searchApis(eq(GraviteeContext.getExecutionContext()), any(), any(), any());
+
+        final Response response = target("/_search")
+            .queryParam("q", "3")
+            .queryParam("categoryId", "12345")
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/ApisResourceTest.java]: why categoryId? Query param is called category

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java [hunk 3]
```diff
@@ -313,6 +321,11 @@ public List<ApiCriteria> computeApiCriteriaForUser(
                 apiCriteriaList.add(queryToCriteria(executionContext, apiQuery).ids(userGroupApiIds).build());
             }
 
+            if (!isBlank(apiQuery.getCategory())) {
+                var categoryName = categoryService.findById(apiQuery.getCategory(), executionContext.getEnvironmentId()).getKey();
```
Comment (wbabyte) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java:325]: why we had this for current implementation while building query
`builder.category(categoryService.findById(query.getCategory(), executionContext.getEnvironmentId()).getId())`
while in db we keep the name of the category, not id 
did you found the answer? Maybe there is a bug?

**Review by wbabyte — COMMENTED** [review]
few remarks to discuss. 
also remember to add Jira ticket and description 
use conventional commits and squash all of your commits to one

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java (hunk 3 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java:325]: In Apis table we keep not categories ids, but they keys.
That is why we cannot search category by id, but by it's key instead

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java (hunk 1 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java:125]: I think I will discuss with @phiz71 where the category parameter should be

File: gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/ApisResourceTest.java (hunk 2 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/test/java/io/gravitee/rest/api/portal/rest/resource/ApisResourceTest.java]: Updated

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java (hunk 3 above)
Comment (wbabyte) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java:325]: I know, but I saw in code places that set api query category to id and not the name like you did

**Pull request closed without merging**